Carousel: disable UI controls for single image

ramonjd opened the pull request:
Fixes https://github.com/Automattic/jetpack/issues/20267

#### Changes proposed in this Pull Request

This PR:

- hides navigation arrows and pagination UI if there's only one image.
- disables swipe interaction if there's only one image.
- sets swiper to `enabled` before close. We have to do this because, after testing, it seems as if Swiper doesn't re-enable the navigation arrows on reinitialization.


**Before**

<img width="500" alt="single-image" src="https://user-images.githubusercontent.com/6458278/127795393-aed572ba-fe2c-4523-8c5e-9f7390b5fd0d.png">

**After**
<img width="500" alt="Screen Shot 2021-08-02 at 12 10 10 pm" src="https://user-images.githubusercontent.com/6458278/127795386-558356cd-611c-4ae1-8683-089589e1abe1.png">

#### Does this pull request change what data or activity we track or use?
Nope.

#### Testing instructions:

Create 2 x galleries on the same post: one with several images and one with only 1 image.

Publish them and checkout the frontend.

You should see and be able to use the navigation arrows and pagination UI for the gallery with several images.

The gallery with one image shouldn't display any UI controls. You shouldn't be able to slide it or interact with the image.

Open one, then the other to make sure we're enabling/disabling the UI correctly.

ramonjd committed:
[not verified] Hiding pagination and nav navigation controls if there is only one image in the slides array.
Displaying them again before we close the carousel to ensure they display for other galleries in the same page.

andrewserong commented on projects/plugins/jetpack/modules/carousel/jetpack-carousel.js:
```diff
@@ -472,6 +479,10 @@
 					// Fixes some themes where closing carousel brings view back to top.
 					document.documentElement.style.removeProperty( 'height' );
 
+					// Should the navigation arrows if we hid them for single images.
```
```suggestion
					// Show the navigation arrows if we hid them for single images.
```
Tiny typo 🙂

ramonjd replied:
By blod is typo


😄

ramonjd committed:
[not verified] Update projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

Co-authored-by: Andrew Serong <14988353+andrewserong@users.noreply.github.com>

andrewserong reviewed (dismissed):
Nice change @ramonjd, hides the next/prev buttons and the lone dot in the bottom left 👍 

I noticed there's a very slight flicker where you can see the next/prev buttons for a moment when the carousel is opened, and then again when it's closed:

![carousel-single-image-arrows-sml](https://user-images.githubusercontent.com/14988353/127816721-f637076c-00d4-4c40-9d7b-bda3264bef06.gif)

It's subtle, though, and I suspect most people won't notice, so not a blocker for getting this in.

Tested in Chrome & Safari on iOS and desktop, and FF and Edge on desktop — no lone pagination circle, and no next/prev buttons on a single image in an image gallery. Pagination and next/prev buttons correctly display on a bigger gallery on the same page, and opening/closing between the two works as expected 👍

LGTM!

ramonjd committed:
[not verified] Using swiper to enable/disable

ramonjd commented:
@andrewserong Thanks a lot for the review! I think I hadn't pushed the changes I'd intended. 🤕 

I wanted to control the nav + interaction using the Slide API. I've updated the PR and rebased. Hopefully it'll take care of the flashing nav buttons too.

If you have time for another quick review, I'd appreciate it!!

andrewserong commented:
> If you have time for another quick review, I'd appreciate it!!

Sure thing! Thanks Ramon, I'll give this a whirl later on this morning.

ramonjd committed:
Changing capitalization in comment sentence

"Down with capitalization!" - Karl 'puncutation' Marx

andrewserong reviewed (approved):
This is working nicely for me @ramonjd, re-tested on iOS and desktop browsers on Mac. There's no longer a flash of the next/prev buttons when the carousel is opened. There's still a flash for me when the carousel is closed (presumably because we have to call `swiper.enable()` in `beforeClose` so that it's called before `swiper.destroy()` in `afterClose`. This seems totally reasonable to me, it's such a minor thing, and probably not worth adding in extra CSS / `domUtil.hide` / `show`s just for that case.

LGTM! 🎉

andrewserong commented:
@ramonjd looks like this one needs a `git merge origin/master` before it can be merged 🙂

ramonjd commented:
> @ramonjd looks like this one needs a git merge origin/master before it can be merged 🙂

Thanks for heads up. I'll try to get it in soon before more changes arrive! 😄

ramonjd committed:
Merge remote-tracking branch 'origin/master' into update/carousel-hide-ui-controls-for-single-image

ramonjd commented:
r229791-wpcom
